### PR TITLE
Allow user to control output of verbose messages

### LINKED ISSUE #52244: ansible-inventory mixes traceback with stdout when set to verbose
##### SUMMARY

Traceback from an inventory plugin is mixed with stdout instead of being directed to stderr when using `ansible-inventory -vvvv`. This is a blocker for awx being able to seperate the streams and correctly display tracebacks to users.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

ansible-inventory
##### ANSIBLE VERSION

```paste below
ansible-inventory 2.8.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/elijah/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/elijah/sfw/ansible/ansible/lib/ansible
  executable location = /home/elijah/envs/py36env/bin/ansible-inventory
```

##### CONFIGURATION

```paste below
 ansible-config dump --only-changed
ANSIBLE_COW_WHITELIST(/etc/ansible/ansible.cfg) = ['bud-frogs', 'bunny', 'cheese', 'daemon', 'default', 'dragon', 'el>
DEFAULT_LOAD_CALLBACK_PLUGINS(/etc/ansible/ansible.cfg) = True
DEFAULT_STDOUT_CALLBACK(/etc/ansible/ansible.cfg) = yaml
INVENTORY_ENABLED(/etc/ansible/ansible.cfg) = ['auto', 'ini', 'script', 'host_list', 'virtualbox', 'yaml', 'construct>
```

##### OS / ENVIRONMENT

Fedora 29

##### STEPS TO REPRODUCE

Simulate a runtime error by inserting a `raise AnsibleError('message of your choice')` in a plugin. For example:
```diff
diff --git a/lib/ansible/plugins/inventory/aws_ec2.py b/lib/ansible/plugins/inventory/aws_ec2.py
index e834a9078b..bdc0332c34 100644
--- a/lib/ansible/plugins/inventory/aws_ec2.py
+++ b/lib/ansible/plugins/inventory/aws_ec2.py
@@ -371,6 +371,7 @@ class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
         all_instances = []
 
         for connection, region in self._boto3_conn(regions):
+            raise AnsibleError('Something blew up')
             try:
                 # By default find non-terminated/terminating instances
                 if not any([f['Name'] == 'instance-state-name' for f in filters]):

```
> Note: this could be any plugin

Then, call this plugin with a valid yaml file for that plugin with high verbosity, directing stderr to `/dev/null`
```sh
ansible-inventory -vvvv -i aws_ec2.yml --list 2>/dev/null
```

##### EXPECTED RESULTS

I only get stdout from the plugin if I direct stderr to somewhere else.

##### ACTUAL RESULTS

I get traceback mixed in with sdtout

```sh
ansible-inventory -vvvv -i aws_ec2.yml --list 2>/dev/null
ansible-inventory 2.8.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/elijah/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/elijah/sfw/ansible/ansible/lib/ansible
  executable location = /home/elijah/envs/py36env/bin/ansible-inventory
  python version = 3.6.8 (default, Jan 27 2019, 09:00:23) [GCC 8.2.1 20181215 (Red Hat 8.2.1-6)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
script declined parsing /home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.yml as it did not pass it's verify_file() method
host_list declined parsing /home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.yml as it did not pass it's verify_file() method
virtualbox declined parsing /home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.yml as it did not pass it's verify_file() method
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/auto.py", line 55, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 549, in parse
    results = self._query(regions, filters, strict_permissions)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 446, in _query
    return {'aws_ec2': self._get_instances_by_region(regions, filters, strict_permissions)}
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 374, in _get_instances_by_region
    raise AnsibleError('Something blew up')

  File "/home/elijah/sfw/ansible/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/ini.py", line 137, in parse
    raise AnsibleParserError(e)

  File "/home/elijah/sfw/ansible/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/yaml.py", line 112, in parse
    raise AnsibleParserError('Plugin configuration YAML file, not YAML inventory')

  File "/home/elijah/sfw/ansible/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/constructed.py", line 99, in parse
    self._read_config_data(path)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/__init__.py", line 206, in _read_config_data
    raise AnsibleParserError("Incorrect plugin name in file: %s" % config.get('plugin', 'none found'))

  File "/home/elijah/sfw/ansible/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 549, in parse
    results = self._query(regions, filters, strict_permissions)
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 446, in _query
    return {'aws_ec2': self._get_instances_by_region(regions, filters, strict_permissions)}
  File "/home/elijah/sfw/ansible/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 374, in _get_instances_by_region
    raise AnsibleError('Something blew up')

{
    "_meta": {
        "hostvars": {}
    },
    "all": {
        "children": [
            "ungrouped"
        ]
    }
}
```

### Related enhancements:
https://github.com/ansible/ansible/issues/49657

While the above could possibly solve this, and should consider this case, it would be better for this bug to get solved in the short term by a more targeted patch.

### PR DESCRIPTION
**Pull request by bcoca**
to stdout (default) or stderr
fixes #52244

##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

display

### CODE REVIEW

**Comment by bcoca** [comment]
normal:
```
bcoca@erebos:~/work/ansible(allow_verbose_stderr)$ ansible -v -m ping localhost
Using /etc/ansible/ansible.cfg as config file
[WARNING]: No inventory was parsed, only implicit localhost is available

localhost | SUCCESS => {
"changed": false,
"ping": "pong"
}
```
redirecting stderr with False:
```
bcoca@erebos:~/work/ansible(allow_verbose_stderr)$ ansible -v -m ping localhost 2>/dev/null
Using /etc/ansible/ansible.cfg as config file
localhost | SUCCESS => {
"changed": false,
"ping": "pong"
}
```

redirecting stderr with True:
```
bcoca@erebos:~/work/ansible(allow_verbose_stderr)$ ANSIBLE_VERBOSE_TO_STDERR=1 ansible -v -m ping localhost 2>/dev/null
localhost | SUCCESS => {
"changed": false,
"ping": "pong"
}
```

File: lib/ansible/utils/display.py [hunk 1]
```diff
@@ -201,11 +201,13 @@ def debug(self, msg, host=None):
                 self.display("%6d %0.5f [%s]: %s" % (os.getpid(), time.time(), host, msg), color=C.COLOR_DEBUG)
 
     def verbose(self, msg, host=None, caplevel=2):
```
Comment (sivel) [inline lib/ansible/utils/display.py:203]: Would it be useful to allow `verbose` to accept `stderr` in the function signature too?  We would also have to update `v`, `vv`, etc to support it also.

**Comment by sivel** [comment]
On related note, unrelated to my review comment, weren't you fundamentally against this as an option when we had discussed it? I remember you saying that verbose messages aren't errors, and thus shouldn't be sent to stderr.  Just me being curious.

File: lib/ansible/utils/display.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/utils/display.py:203]: yep, but thought that would be huge undertaking while this was a 'blanket' requriement

**Comment by bcoca** [comment]
@sivel still against it, but that was making it mandatory and the default, as a toggle, it bothers me much less

**Comment by AlanCoding** [comment]
From a prior discussion:

https://github.com/ansible/ansible/issues/49657#issuecomment-461475449

> Note that in my original plan, the first phase would be instead of porting display.warning, `display.v*`, etc over to a logging library, it would direct those calls to stderr instead. That first phase would probably be enough to satisfy your use case as well as the json output would be on stdout and the logs would be on stderr.

It looks like this is basically implementing that, but adding a setting to toggle it.

----

Confirmed with

```
ANSIBLE_VERBOSE_TO_STDERR=true ansible-inventory -i private/ec2/aws_ec2.yaml --list --export -vvvvv 2> /dev/null
```

that this does do what is advertised, it does not show debugging statements.

**Comment by bcoca** [comment]
debug is not tied to verbosity, diff animal

**Comment by AlanCoding** [comment]
Okay, yes, I understand. "debug" has its own special meaning and it's not a consideration here. I don't know what to call this output. Config information is written whenever some level of `-v`s are used, and we also need that level of verbosity to get tracebacks. This change does allow redirecting the output to stderr.

**Review by AlanCoding — APPROVED** [review]
In the comments, I posted confirmation that this solves the problem for us with ansible-inventory use. I see no reason to hold this up. Ping @ryanpetrello @matburt, you may want to review the details here.

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**